packages/components/src/Card/Card.tsx

```diff
@@ -55,15 +85,45 @@ export function Card({
 
   const cardContent = (
     <>
-      {title && (
-        <div className={styles.header}>
-          <Heading level={3}>{title}</Heading>
-        </div>
-      )}
+      {CardHeader && CardHeader()}
       {children}
     </>
   );
 
+  function CardHeader() {
+    // Case 1: Deprecated Title
+    if (title) {
+      return (
+        <div className={styles.header}>
+          {title && <Heading level={3}>{title}</Heading>}
+        </div>
+      );
+    }
+    // Case 2: String Header
+    if (typeof header === "string") {
+      return (
+        <div className={styles.header}>
+          {header && <Heading level={3}>{header}</Heading>}
+        </div>
+      );
+    }
+    // Case 3: Custom Header
+    if (React.isValidElement(header)) {
+      return header;
+    }
+    // Case 4: Default Header Props
+    return (
+      !!header && (
+        <div className={styles.header}>
+          {header?.title && <Heading level={3}>{header?.title}</Heading>}
+          {header?.action && (
+            <Button {...(header?.action as ButtonProps)} size="small" />
```

Comment:
Is it possible to _default_ the type to `tertiary`? I'd prefer not to make `primary` the default treatment as you could quickly build out a pretty overwhelming view with >1 cards with primary actions.

Reply:
added default Button type to be `tertiary` if it's not specified in the header action props. Not sure if we need to mention this level of details in the documentation.